registry/txt.go

```diff
@@ -397,3 +407,19 @@ func (im *TXTRegistry) removeFromCache(ep *endpoint.Endpoint) {
 		}
 	}
 }
+
+func (im *TXTRegistry) filterExistingRecords(records []*endpoint.Endpoint) []*endpoint.Endpoint {
+	filtered := []*endpoint.Endpoint{}
+	for _, r := range records {
+		if r.RecordType != endpoint.RecordTypeTXT {
+			filtered = append(filtered, r)
+			continue
+		}
+		for _, t := range im.txtRecordsCache {
```

Comment:
Should the `txtRecordsCache` just be a map with the key being the identifier and the DNSName since we are using it like this? Would make lookups quicker and avoid this inner loop.

Reply:
Good idea, that will speed things up.

The lint error is from my previous PR actually, I will run go fmt on it afterwards.